utils/mermaid.js

```diff
@@ -0,0 +1,192 @@
+	openNode.length = 0;
+	callFlow.length = 0;
+	showHideDesc = "";
+	baseGraph = "";
+};
+
+var drawDepthGraph = function(depth) {
+	mermaid.initialize(config);
+	showHideDesc = "";
+	openNode.length = 0;
+	const nodeList = new Array();
+	let id = new String();
+	for (let i = 0; i < callFlow.length; i++) {
+		callFlow[i] = callFlow[i].replace(COMMENT, "");
+		if (Number(startDepth[i]) >= depth)
+			callFlow[i] = (COMMENT + callFlow[i]);
+		if (Number(startDepth[i]) == depth) {
+			id = startDepth[i] + "_" + startId[i];
+			if (nodeList.includes(id) == false) {
+				nodeList.push(id);
+				showHideDesc = showHideDesc.concat(
+					"  style " + id +
+					" fill:#f9f,stroke:#333,stroke-width:px;\n");
+				showHideDesc =
+					showHideDesc.concat("  click " + id + " showHideNode;\n");
+			}
+		}
+	}
+	renderGraph("".concat(GRAPH_TITLE, callFlow.join("\n"), "\n", showHideDesc));
+};
+
+var drawSelectGraph = function() {
+	let selectionDesc = new String();
+	let nodeList = new Array();
+	if (callFlow.length == 0)
+		drawBaseGraph(0);
+	for (let i = 1; i < callFlow.length; i++) {
+		let id = startDepth[i] + "_" + startId[i];
+		if (nodeList.includes(id) == false) {
+			nodeList.push(id);
+			selectionDesc = selectionDesc.concat(
+				"  style " + id + " fill:#CAB8FF,stroke:#333,stroke-width:px;\n");
+			selectionDesc =
+				selectionDesc.concat("  click " + id + " selectSubGraph;\n");
+		}
+	}
+	renderGraph(baseGraph.concat("\n", selectionDesc));
+	document.getElementById("description").innerHTML = "Click on a node to view Sub-Graph";
+};
+
+var drawBaseGraph = function(id) {
+	mermaid.initialize(config);
+	initializeBaseGraph();
+	let start = new Boolean(false);
+	maxDepth = 0;
+	var result = mermaidTxt.split(/\r?\n/);
+	for (let i = 0; i < result.length; i++) {
+		if (result[i].length == 0)
+			continue;
+		let oneCallFlow = result[i].match(
+			/\s+(?<start_depth>\d+)_(?<start_id>\d+)\[\"(?<start_name>\S+)\"\]\s+--\S+\|(?<call_num>\d+)\|\s+(?<end_depth>\d+)_(?<end_id>\d+)\[\"(?<end_name>\S+)\"];/);
+		if (oneCallFlow == null)
+			continue;
+		if (Number(oneCallFlow.groups.start_id) == id) {
+			start = true;
+			baseRootDepth = Number(oneCallFlow.groups.start_depth);
+			baseRootId = id;
+		}
+		if (start == true) {
+			if (Number(oneCallFlow.groups.start_depth) > baseRootDepth ||
+			    Number(oneCallFlow.groups.start_id == id)) {
+				callFlow.push(result[i]);
+				startId.push(oneCallFlow.groups.start_id);
+				startDepth.push(oneCallFlow.groups.start_depth);
+				startName.push(oneCallFlow.groups.start_name);
+			}
+			else {
+				break;
+			}
+		}
+		if (Number(oneCallFlow.groups.start_depth) > maxDepth)
+			maxDepth = Number(oneCallFlow.groups.start_depth);
+	}
+	baseGraph = "".concat(GRAPH_TITLE, callFlow.join("\n"));
+	updateDepthList();
+	renderGraph(baseGraph);
+};
+
+var updateDepthList =
+	function() {
+	var select = document.getElementById("depth-select");
+	select.options.length = 0;
+	select.options[0] = new Option("Full", 64);
```

Comment:
FULL_DEPTH ?

Reply:
Yes. I updated it to use the defined constant.
Thank you for the review.